include/phasar/PhasarLLVM/Utils/OnTheFlyAnalysisPrinter.h

```diff
@@ -0,0 +1,50 @@
+#ifndef PHASAR_PHASARLLVM_UTILS_ONTHEFLYANALYSISPRINTER_H
+#define PHASAR_PHASARLLVM_UTILS_ONTHEFLYANALYSISPRINTER_H
+
+#include "phasar/Domain/BinaryDomain.h"
+#include "phasar/PhasarLLVM/Utils/AnalysisPrinterBase.h"
+#include "phasar/Utils/IO.h"
+#include "phasar/Utils/MaybeUniquePtr.h"
+#include "phasar/Utils/Printer.h"
+
+#include <cassert>
+
+#include <llvm/ADT/Twine.h>
+#include <llvm/Support/raw_ostream.h>
+namespace psr {
+
+template <typename AnalysisDomainTy>
+class OnTheFlyAnalysisPrinter : public AnalysisPrinterBase<AnalysisDomainTy> {
+  using l_t = typename AnalysisDomainTy::l_t;
+
+public:
+  explicit OnTheFlyAnalysisPrinter(llvm::raw_ostream &OS)
+      : AnalysisPrinterBase<AnalysisDomainTy>(), OS(&OS){};
+
+  explicit OnTheFlyAnalysisPrinter<AnalysisDomainTy>(
+      const llvm::Twine &Filename)
+      : AnalysisPrinterBase<AnalysisDomainTy>(), OS(openFileStream(Filename)){};
+
+  OnTheFlyAnalysisPrinter<AnalysisDomainTy>() = default;
+  ~OnTheFlyAnalysisPrinter<AnalysisDomainTy>() = default;
+
+  void onInitialize() override{};
+  void onResult(Warning<AnalysisDomainTy> Warn) override {
+    assert(isValid());
```

Comment:
If you are only using **isValid** in the assert statement, you can move the condition directly into the assert and remove the function.

Reply:
Initial thoughts was to keep the **isValid** as a function so that the client analyses can use this to check whether OS is not nullptr before calling **onResult** since **openFileStream** could return a nullptr.

Reply:
Then, we can default the openFileStream to llvm::errs() and remove the isValid function.